src/actions/write/buyRecord.ts

```diff
@@ -42,23 +66,20 @@ export const buyRecord = async (
     throw new ContractError(`Caller balance is not defined!`);
   }
 
-  // Check if it includes a valid number of years
-  if (!Number.isInteger(years) || years > MAX_YEARS || years <= 0) {
+  // Additional check if it includes a valid number of years (TODO: this may be set in contract settings)
+  if (years > MAX_YEARS || years <= 0) {
```

Comment:
So you're just leaving this in place at the moment while we figure out how to square up contract-state-based setting vs. schema validator setting?

Reply:
yeah, if we set these in the state - I'm not sure we'd want to add them to the schema `minimum/maximum`